frontend/src/lib/cipherly.ts

```diff
@@ -159,6 +159,32 @@ export function decodePayload(data: string): Payload {
   return decodeMessagePack(extractHash(data)) as Payload;
 }
 
+export async function passwordDecrypt(
```

Comment:
(1) I moved the decrypt logic out of `auth.svelte` and `password.svelte` and moved them initially to `/auth/+page.svelte`. But this file got longer so moved it to `cipherly.ts`.


I did this (1), because the way a generic `decrypt` function was bound depending on auth vs password did not work as intended with svelte reactivity and caused some weird issues.

I hope this refactoring is ok, that's the only way I was able to make it work.

Reply:
Yes, this is fine. Thanks.